google/cloud/datastore/key.py

```diff
@@ -441,7 +441,9 @@ def id_or_name(self):
         :returns: The last element of the key's path if it is either an ``id``
                   or a ``name``.
         """
-        return self.id or self.name
+        if self.id is None:
+            return self.name
+        return self.id
```

Comment:
Please add a unit test exercising #106 (i.e., set the key's `id` to 0 and assert that `id_or_name` returns the 0).